mesa/model.py

```diff
@@ -41,6 +41,29 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self.running = True
         self.schedule = None
         self.current_id = 0
+        self.agents_by_type = {}
+
+    def create_agent(self, agent_type, *args, **kwargs):
```

Comment:
@rht Could we type hint that `agent_type` should be an instance of Agent? How would that work?

Reply:
If I am not mistaken it should be the same as in the agent class where you pass in the model `model: Model` so it would be `agent:Agent` you will need to import the Agent module though.